pkgs/applications/window-managers/sway/default.nix

```diff
@@ -34,7 +27,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     wayland libxkbcommon pcre json_c dbus libevdev
     pango cairo libinput libcap pam gdk_pixbuf
-    wlroots wayland-protocols
+    wlroots wayland-protocols swaybg
```

Comment:
Have you managed to test this as it was my understanding that this would not be sufficient to get sway to find swaybg

Reply:
The following works, but it feels like a rushed solution to #60434.

```diff
diff --git a/nixos/modules/programs/sway.nix b/nixos/modules/programs/sway.nix
index b4f03151cdc..f92d09a7ef4 100644
--- a/nixos/modules/programs/sway.nix
+++ b/nixos/modules/programs/sway.nix
@@ -55,7 +55,7 @@ in {
     extraPackages = mkOption {
       type = with types; listOf package;
       default = with pkgs; [
-        swaylock swayidle
+        swaylock swayidle swaybg
         xwayland rxvt_unicode dmenu
       ];
       defaultText = literalExample ''
diff --git a/pkgs/applications/window-managers/sway/default.nix b/pkgs/applications/window-managers/sway/default.nix
index e2f9786bf4e..a6dd89fd90d 100644
--- a/pkgs/applications/window-managers/sway/default.nix
+++ b/pkgs/applications/window-managers/sway/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, fetchpatch
+{ stdenv, fetchFromGitHub, fetchpatch, makeWrapper
 , meson, ninja
 , pkgconfig, scdoc
 , wayland, libxkbcommon, pcre, json_c, dbus, libevdev
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     wayland libxkbcommon pcre json_c dbus libevdev
     pango cairo libinput libcap pam gdk_pixbuf
-    wlroots wayland-protocols swaybg
+    wlroots wayland-protocols swaybg makeWrapper
   ];
 
   enableParallelBuilding = true;
@@ -37,6 +37,10 @@ stdenv.mkDerivation rec {
     "-Dtray=enabled" "-Dman-pages=enabled"
   ];
 
+  postInstall = ''
+    wrapProgram $out/bin/sway --prefix PATH : "${swaybg}/bin"
+  '';
+
   meta = with stdenv.lib; {
     description = "i3-compatible tiling Wayland compositor";
     homepage    = https://swaywm.org;
```

Reply:
I think this is essentially option 1 from #60434

Reply:
Yes, you're right, I included that by accident.

Reply:
There also seems to be a swaybg_command argument for sway though I can't find much documentation on it

Reply:
Maybe add a `programs.sway.swaybg_command`, that defaults to `"${swaybg}/bin/swaybg"`?

Reply:
and add this as an argument in the existing sway wrapper?

Reply:
Yes

Reply:
A quick solution for the meantime could be to patch Sway with another default for `swaybg`.

I.e. use `"${swaybg}/bin/swaybg` here:
https://github.com/swaywm/sway/blob/1.1.1/sway/config.c#L286

That would work with both Nix and NixOS and we could get away without changing the module for now, until we find a proper solution (there are also other packages like `swaylock`, `swayidle`, ... that should be handled in the same way as `swaybg` IMO).

Reply:
What reasoning is there for patching sway when it already provides a way to pass in a program for the background?

Reply:
> What reasoning is there for patching sway when it already provides a way to pass in a program for the background?

We only modify the default value, which we can do in the source code, but we cannot modify the configuration files of existing users, if they've copied them to $HOME, etc.

> What reasoning is there for patching sway when it already provides a way to pass in a program for the background?

Update: But unfortunately something like this doesn't seem to exist (AFAIK).

I also forgot about `makeWrapper` so let's probably go with:
```
postInstall = ''
  wrapProgram $out/bin/sway --prefix PATH : "${swaybg}/bin"
'';
```
to avoid yet another patch for now.

Reply:
> Update: But unfortunately something like this doesn't seem to exist (AFAIK).

Doesn't https://github.com/swaywm/sway/blob/master/sway/commands/swaybg_command.c pull in a custom command from the command line or am I just confused

Reply:
I think I was confused, however if upstream sway doesn't provide a way to set the background program then surely we shouldn't either